src/posix/platform/multithread_api.cpp

```diff
@@ -0,0 +1,260 @@
+
+    platformConfig.mInterfaceName = iface;
+    VerifyOrDie(platformConfig.mRadioUrlNum < OT_ARRAY_LENGTH(platformConfig.mRadioUrls), OT_EXIT_INVALID_ARGUMENTS);
+    platformConfig.mRadioUrls[platformConfig.mRadioUrlNum++] = radioUrl;
+#if defined(__linux)
+    platformConfig.mRealTimeSignal = 41;
+#endif
+    platformConfig.mSpeedUpFactor = 1;
+
+    syslog(LOG_INFO, "Running %s", otGetVersionString());
+    syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
+    IgnoreError(otLoggingSetLevel(config->mOtLogLevel));
+
+    gInstance = otSysInit(&platformConfig);
+    VerifyOrDie(gInstance != NULL, OT_EXIT_FAILURE);
+    syslog(LOG_INFO, "Thread interface: %s", otSysGetThreadNetifName());
+    syslog(LOG_INFO, "RCP version: %s", otPlatRadioGetVersionString(gInstance));
+
+    otLogDebgPlat("ot instance create success!!!");
+
+    pthread_cond_signal(&gCond);
+    pthread_mutex_unlock(&gLock);
+
+    while (gTerminate == false)
+    {
+        otSysMainloopContext mainloop;
+
+        otTaskletsProcess(gInstance);
+
+        FD_ZERO(&mainloop.mReadFdSet);
+        FD_ZERO(&mainloop.mWriteFdSet);
+        FD_ZERO(&mainloop.mErrorFdSet);
+
+        mainloop.mMaxFd           = -1;
+        mainloop.mTimeout.tv_sec  = 10;
+        mainloop.mTimeout.tv_usec = 0;
+
+        otSysMainloopUpdate(gInstance, &mainloop);
+
+        if (otSysMainloopPoll(&mainloop) >= 0)
+        {
+            pthread_mutex_lock(&gLock);
+
+            otSysMainloopProcess(gInstance, &mainloop);
+
+            pthread_mutex_unlock(&gLock);
+        }
+        else if (errno != EINTR)
+        {
+            perror("select");
+            ExitNow();
+        }
+    }
+
+exit:
+    otSysDeinit();
+    gThreadId  = 0;
+    gInstance  = NULL;
+    gTerminate = false;
+    syslog(LOG_INFO, "terminate thread mainloop : exit");
+    return nullptr;
+}
+
+void otSysGetInstance(otInstance **aInstance, const char *aComPort, uint16_t aOtLogLevel)
+{
+    OtConfig *config = nullptr;
+
+    syslog(LOG_INFO, "otSysGetInstance");
+
+    if (gInstance)
+    {
+        syslog(LOG_INFO, "ot instance already initialised!!!");
+        *aInstance = gInstance;
+        ExitNow();
+    }
+
+    config = (OtConfig *)malloc(sizeof(OtConfig));
+    VerifyOrExit(config != nullptr, syslog(LOG_INFO, "malloc failed"));
+    memset(config, 0, sizeof(OtConfig));
+    strncpy(config->mComPort, aComPort, strlen(aComPort));
+    config->mOtLogLevel = aOtLogLevel;
+
+    VerifyOrExit(pthread_mutex_init(&gLock, NULL) == 0, syslog(LOG_INFO, "mutex init has failed"));
+
+    VerifyOrExit(pthread_cond_init(&gCond, NULL) == 0, syslog(LOG_INFO, "cond init has failed"));
+
+    pthread_create(&gThreadId, NULL, pThreadOtMainLoop, config);
+    pthread_mutex_lock(&gLock);
+
+    syslog(LOG_INFO, "Wait for signal to initialise openthread stack !!!");
+    pthread_cond_wait(&gCond, &gLock);
+
+    *aInstance = gInstance;
+
+    otLogDebgPlat("otThread : gThreadId[%ld]", gThreadId);
```

Comment:
If `otSysInit` fails, this log will not printed.
How about we use `syslog`, instead of `otLogDebgPlat` ?
->
syslog(LOG_INFO, "otThread : gThreadId[%ld]", gThreadId);

Reply:
Updated